WooCommerce/Classes/ViewRelated/Dashboard/Settings/Plugins/PluginListViewModel.swift

```diff
@@ -1,13 +1,108 @@
 import Foundation
 import Yosemite
+import protocol Storage.StorageManagerType
 
 final class PluginListViewModel {
 
+    /// ID of the site to load plugins for
+    ///
+    private let siteID: Int64
+
     /// Reference to the StoresManager to dispatch Yosemite Actions.
     ///
     private let storesManager: StoresManager
 
-    init(storesManager: StoresManager = ServiceLocator.stores) {
+    /// StorageManager to load plugins from storage
+    ///
+    private let storageManager: StorageManagerType
+
+    /// Results controller for the plugin list
+    ///
+    private lazy var resultsController: ResultsController<StorageSitePlugin> = {
+        let predicate = NSPredicate(format: "siteID = %ld", self.siteID)
+        let nameDescriptor = NSSortDescriptor(keyPath: \StorageSitePlugin.name, ascending: true)
+        let statusDescriptor = NSSortDescriptor(keyPath: \StorageSitePlugin.status, ascending: true)
```

Comment:
NSFetchedResultsController requires that if we group data by sections then the first sort descriptor should have the same key path as the `sectionNameKeyPath`, otherwise the sorting will get messed up if we add sort descriptor with different key path. I found a reference for it [here](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CoreData/nsfetchedresultscontroller.html) (this should be more transparent in the NSFetchedResultsController document though).

Reply:
That's probably a good annotation to documented on comments.